### ISSUE TITLE
A question regarding the refiner stage

### ISSUE THREAD

**Issue by sen-mao** [czg1225/CoDe#8]
Amazing work! I have a small question regarding the refiner stage. Since only the logits corresponding to the highest scale (i.e., **logits_BlV[:B, cur_L - pn*pn : cur_L]** in Line754) are needed, why is the entire **logits_BlV** tensor processed in the **for** loop (Line750-752)?

https://github.com/czg1225/CoDe/blob/bfb204de8648e38b174ba29870cbd0b7799d85d6/models/var.py#L750-L754

**Comment by czg1225** [comment]
Hi @sen-mao ,
This is to ensure that the entire random process is the same as the original inference, so that the pictures generated by the CoDe under the same seed are the same as the pictures generated by the original inference process. You can also choose to ignore these during inference, which will not cause quality degradation.

**Comment by sen-mao** [comment]
I'm clear now — thanks for the explanation.
